### PR TITLE
Pass entire text through formatWithShebang() to format()

### PR DESCRIPTION
**Pull request by josephfrazier**
That way, we won't have to do any arithmetic on range and cursor locations
when there is a shebang. See here for details:
                                                                           
https://github.com/prettier/prettier/pull/1637#issuecomment-303846507
                                                                           
This required changing comment printing such that comments that are
actually shebangs are just ignored.

### CODE REVIEW

File: src/comments.js [hunk 1]
```diff
@@ -801,6 +801,9 @@ function printComment(commentPath) {
       return "/*" + comment.value + "*/";
     case "CommentLine":
     case "Line":
+      // Don't print the shebang, it's taken care of in index.js
+      if (options.originalText.slice(util.locStart(comment)).startsWith("#!"))
```
Comment (vjeux) [inline src/comments.js]: please add {} :)

Comment (josephfrazier) [inline src/comments.js]: oops, i'll also add an eslint rule in a separate PR

**Comment by vjeux** [comment]
Oh nice! We used to have this crazy setup because the parsers threw syntax errors with shebang. Glad to know they are now handling it :)

Feel free to merge once you address my small nit

File: src/comments.js (hunk 1 above)
Comment (vjeux) [inline src/comments.js]: Awesome :)

**Comment by josephfrazier** [comment]
Ah, that's neat to know. Ideally, we could handle actually printing the shebang in `printComment`, rather than lopping it off beforehand and reattaching it afterward. However, it looks like the flow/typescript parsers don't include the shebang in their ASTs :/

**Pull request merged**